python/packages/core/agent_framework/_workflows/_group_chat.py

```diff
@@ -0,0 +1,1505 @@
+- final_response: when finish is true, provide the final answer to the user
+"""
+
+
+class _PromptBasedGroupChatManager:
+    """LLM-backed manager that produces directives via structured output.
+
+    This is the default manager implementation for group chat workflows. It uses an LLM
+    to make speaker selection decisions based on conversation state, participant
+    descriptions, and custom instructions.
+
+    Coordination strategy:
+    - Receives immutable state snapshot with full conversation history
+    - Formats system prompt with instructions, task, and participant descriptions
+    - Appends conversation context and structured output prompt
+    - Parses LLM response (JSON mapping) and converts to GroupChatDirective
+
+    Flexibility:
+    - Custom instructions allow domain-specific coordination strategies
+    - Participant descriptions guide the LLM's selection logic
+    - Structured output ensures reliable parsing (no regex or brittle prompts)
+
+    Example coordination patterns:
+    - Round-robin: "Rotate between participants in order"
+    - Task-based: "Select the participant best suited for the current sub-task"
+    - Dependency-aware: "Only call analyst after researcher provides data"
+
+    Args:
+        chat_client: ChatClientProtocol implementation for LLM inference
+        instructions: Custom system instructions (defaults to DEFAULT_MANAGER_INSTRUCTIONS)
+        name: Display name for the manager in conversation history
+
+    Raises:
+        RuntimeError: If LLM response cannot be parsed into the directive payload
+                     If directive is missing next_agent when finish=False
+                     If selected agent is not in participants
+    """
+
+    def __init__(
+        self,
+        chat_client: ChatClientProtocol,
+        *,
+        instructions: str | None = None,
+        name: str | None = None,
+    ) -> None:
+        self._chat_client = chat_client
+        self._instructions = instructions or DEFAULT_MANAGER_INSTRUCTIONS
+        self._name = name or "GroupChatManager"
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    async def __call__(self, state: GroupChatStateSnapshot) -> GroupChatDirective:
+        participants = state["participants"]
+        task_message = state["task"]
+        conversation = state["conversation"]
+
+        participants_section = "\n".join(f"- {agent}: {description}" for agent, description in participants.items())
+
+        system_message = ChatMessage(
+            role=Role.SYSTEM,
+            text=(f"{self._instructions}\n\nTask:\n{task_message.text}\n\nParticipants:\n{participants_section}"),
+        )
+
+        messages: list[ChatMessage] = [system_message, *conversation]
+        messages.append(
+            ChatMessage(
+                role=Role.USER,
+                text=(
+                    "Return a JSON object with keys (next_agent, message, finish, final_response). "
```

Comment:
We should surface the default prompt, maybe as a class variable or through default value for the `set_prompt_based_manager` method.